### ISSUE TITLE
Update OPRF to adapt to POPRF

### ISSUE THREAD

**Issue by bytemare** [cfrg/draft-irtf-cfrg-opaque#281]
The (V)OPRF spec has been updated to include the POPRF findings. We should integrate these changes since this spec depends on it.

_Update: POPRF merged with #282, but some questions remain_

- _Does it make sense to have client-side domain separation in both POPRF (info field) and the AKE (context string)?_
- _If not, what are the arguments to have it on one part or the other?_

**Comment by gtank** [comment]
This adds an important security property for our (@celo-org's) OPAQUE use case. We can't strongly identify or authorize the users of our OPRF endpoint because it's part of an account recovery flow, so we rely on rate limiting client identities to prevent brute force attempts. As current specified OPAQUE seems to allow the following attack (which @nategraf and I spotted here and also in our current design with a different OPRF construction - it's a fundamental issue):

1. Attacker submits a fraudulent request for the credential file of a target identity. They receive the encrypted blob but fail to decrypt it. This preserves the main security guarantees, since it prevents attacker access to the stored credential or a valid session key, but it still gives them a guess at the password's OPRF evaluation. We assume rate limiting stops giving OPRF outputs in response to their password guesses at some point, but the attacker can retain the encrypted credential file of the target account.

2. Attacker then creates a sequence of new accounts and requests their own credential files, but continues guessing the target's password and applying the OPRF outputs to decryption attempts on the target file.

Basically, since the OPRF evaluation isn't domain-separated by identity, then the target's masking key can be derived from a successful guess by anyone. You could use different keys for different users (which is how some POPRFs work anyway) but a single-key POPRF would allow us to bind the client ID to the OPRF evaluation and enforce rate limits on a per-client basis without the complexity of additional keys.

**Comment by chris-wood** [comment]
Overcome by #324.
